aptos-move/aptos-vm/src/move_vm_ext/session/mod.rs

```diff
@@ -358,9 +363,25 @@ where
 
             for (struct_tag, blob_op) in resources {
                 let resource_group_tag = {
+                    // MODULE METERING SAFETY:
+                    //   If this resource is in data cache, we must have already fetched metadata
+                    //   for its tag. Note that we do this check for lazy loading only because the
+                    //   eager loader was never traversing modules loaded due to metadata accesses.
```

Comment:
I'm not sure I understand the comment here, this is also an invariant check for lazy loading?

Reply:
Yes, updated to `// INVARIANT:`

Technically, this metadata can be extracted directly from data cache, but to keep things simple and minimal, I did not do it. Having invariant check here ensures we cover all possible accesses for now.